dnsimple/templates.go

```diff
@@ -128,6 +128,22 @@ func (s *TemplatesService) DeleteTemplate(accountID string, templateID string) (
 	return templateResponse, nil
 }
 
+// ApplyTemplate applies a template to the given domain.
+//
+// See https://developer.dnsimple.com/v2/templates/domains/#apply
+func (s *TemplatesService) ApplyTemplate(accountID string, domainID string, templateID string) (*TemplateResponse, error) {
```

Comment:
@jodosha people using the client ignores the API path. And in fact, they don't care.

If we follow the path, we should then define this method under domains, not templates (which I prefer not).

Let's take the case we have both: one in domains, and one in template. The one in domains delegates to template.

Each method should have a different signature that reflects the scope where it is used. It is really confusing to me that all the methods in template that deal with a template start with `account, template, else` and here we do the opposite.